changelogs/fragments/7158-gitlab-project-default-branch-update.yml

```diff
@@ -0,0 +1,2 @@
+breaking_changes:
+  - gitlab_project - add ``default_branch`` support for project update (https://github.com/ansible-collections/community.general/pull/7158).
```

Comment:
```suggestion
  - gitlab_project - add ``default_branch`` support for project update. If you used the module so far with ``default_branch`` to update a project, the value of ``default_branch`` was ignored. Make sure that you either do not pass a value if you are not sure whether it is the one you want to have to avoid unexpected breaking changes (https://github.com/ansible-collections/community.general/pull/7158).
```